### PR TITLE
Bump vendored deps and add maintenance guide to `src_ext`

### PR DESCRIPTION
**Pull request by dra27**

### CODE REVIEW

**Comment by dra27** [comment]
Oops, apparently I forgot about CI building in dev mode 🙈 The update to cmdliner 1.1.0 is optional (at this stage) - the main reason was I noticed the incorrect requirement for `m4` still in the Windows part of the README.

**Comment by hannesm** [comment]
FWIW, I'd prefer to update to dune 3.0.3 directly due to https://github.com/ocaml/dune/commit/a6bc3ee17d3af36b063bec993b5ba10ff4699945 (and the getconf thingy) -- otherwise it is broken on FreeBSD and OpenBSD.

File: src_ext/Makefile.sources [hunk 1]
```diff
@@ -18,8 +18,8 @@ MD5_re = a36347dcfaf71c95916f96f72b0cf2ce
 
 $(call PKG_SAME,re)
 
-URL_cmdliner = http://erratique.ch/software/cmdliner/releases/cmdliner-1.0.4.tbz
-MD5_cmdliner = fe2213d0bc63b1e10a2d0aa66d2fc8d9
+URL_cmdliner = https://erratique.ch/software/cmdliner/releases/cmdliner-1.1.0.tbz
+MD5_cmdliner = 92ec761ff46981239c99a3daead60ec7
```
Comment (rjbou) [inline src_ext/Makefile.sources:22]:
```suggestion
URL_cmdliner = http://erratique.ch/software/cmdliner/releases/cmdliner-1.0.4.tbz
MD5_cmdliner = fe2213d0bc63b1e10a2d0aa66d2fc8d9
```

File: src_ext/Makefile.sources [hunk 2]
```diff
@@ -64,19 +64,19 @@ MD5_result = 1b82dec78849680b49ae9a8a365b831b
 $(call PKG_SAME,result)
 
 # NB If minimum OCaml version for Dune changes, update DUNE_SECONDARY in configure.ac
-URL_dune-local = https://github.com/ocaml/dune/releases/download/2.9.1/dune-2.9.1.tbz
-MD5_dune-local = 0929081f80064ecb836ffdef983100f4
+URL_dune-local = https://github.com/ocaml/dune/releases/download/3.0.2/fiber-3.0.2.tbz
+MD5_dune-local = 8aecdb9b58ab259dd99b52e99a08adb4
```
Comment (rjbou) [inline src_ext/Makefile.sources:68]:
```suggestion
URL_dune-local = https://github.com/ocaml/dune/releases/download/3.0.3/fiber-3.0.3.tbz
MD5_dune-local = 18176405dfcf65d322c029b83754225d
```

**Comment by kit-ty-kate** [comment]
I don’t think this is a good idea to upgrade to dune 3 just yet: https://github.com/ocaml/dune/issues/5455

File: src_ext/Makefile.sources [hunk 3]
```diff
@@ -64,19 +64,19 @@ MD5_result = 1b82dec78849680b49ae9a8a365b831b
 $(call PKG_SAME,result)
 
 # NB If minimum OCaml version for Dune changes, update DUNE_SECONDARY in configure.ac
-URL_dune-local = https://github.com/ocaml/dune/releases/download/2.9.1/dune-2.9.1.tbz
-MD5_dune-local = 0929081f80064ecb836ffdef983100f4
+URL_dune-local = https://github.com/ocaml/dune/releases/download/3.0.3/fiber-3.0.3.tbz
+MD5_dune-local = 18176405dfcf65d322c029b83754225d
```
Comment (rjbou) [inline src_ext/Makefile.sources:68]:
```suggestion
URL_dune-local = https://github.com/ocaml/dune/releases/download/2.9.1/dune-2.9.1.tbz
MD5_dune-local = 0929081f80064ecb836ffdef983100f4
```

**Pull request merged**